### PR TITLE
Updates for gateway deprecation 10/21

### PR DESCRIPTION
**Pull request by randall-Mysten**

### CODE REVIEW

File: doc/src/build/devnet.md [hunk 1]
```diff
@@ -2,11 +2,11 @@
 title: Connect to Sui Devnet
 ---
 
-We are hosting a public Devnet for the community to experiment with Sui and provide feedback to help to shape the future of the Sui network.
+The Sui Devnet network lets you experiment with Sui and provide feedback to help to shape the future of the Sui network.
 
 The Sui Devnet network currently consists of:
 
-* Four validator nodes operated by Mysten Labs. Clients send transactions and read requests via this endpoint: fullnode.devnet.sui.io:443 using [JSON-RPC](../build/json-rpc.md)
+* Four validator nodes operated by Mysten Labs. Clients send transactions and read requests via this endpoint: fullnode.devnet.sui.io:443 using [JSON-RPC](../build/json-rpc.md).
```
Comment (longbowlu) [inline doc/src/build/devnet.md:9]:
```suggestion
* Four validator nodes operated by Mysten Labs. Clients send transactions and read requests via this endpoint: `https://fullnode.devnet.sui.io:443` using [JSON-RPC](../build/json-rpc.md).
```

File: doc/src/build/fullnode.md [hunk 2]
```diff
@@ -1,58 +1,47 @@
 ---
-title: Run a Sui Fullnode
+title: Run a Sui Full node
 ---
 
-Sui full nodes run a service that
-stores the full blockchain state and history. They service reads, either for
-end clients or by helping other full nodes get up-to-date with the latest
-transactions that have been committed to the chain.
+Sui Full nodes run a service that stores the full blockchain state and history. They service reads, either for end clients or by helping other Full nodes get up-to-date with the latest transactions that have been committed to the chain.
```
Comment (longbowlu) [inline doc/src/build/fullnode.md:5]: it's not accurate to say it stores the *full* blockchain data. Also it's worth calling out FN validates blockchain data.
`by helping other Full nodes get up-to-date with the latest transactions that have been committed to the chain.` this is WIP but not yet happening. 

```suggestion
Sui Full nodes run a service that validates blockchain activities, including transactions, checkpoints, and epoch changes etc. It stores and services the queries for the blockchain state and history.
```

File: doc/src/build/fullnode.md [hunk 3]
```diff
@@ -1,58 +1,47 @@
...
 
-This role enables
-[validators](../learn/architecture/validators.md) (or miners in
-other networks) to focus on servicing the write path and processing
-transactions as fast as possible. Once a validator has committed a new set of
-transactions (or a block of transactions), the validator will push that block
-to a full node (potentially a number of full nodes) who will then in turn
-disseminate it to the rest of the network.
+This role enables [validators](../learn/architecture/validators.md) (or miners in
+other networks) to focus on servicing the write path and processing transactions as fast as possible. Once a validator has committed a new set of transactions (or a block of transactions), the validator will push that block to a Full node (potentially a number of Full nodes) who will then in turn disseminate it to the rest of the network.
```
Comment (longbowlu) [inline doc/src/build/fullnode.md:8]: ditto 
```suggestion
This role enables [validators](../learn/architecture/validators.md) (or miners in
other networks) to focus on servicing the write path and processing transactions as fast as possible. Once a validator has committed a new set of transactions (or a block of transactions), the validator will push that block to connected Full nodes who will then service the queries from clients.
```

File: doc/src/build/fullnode.md [hunk 4]
```diff
@@ -1,58 +1,47 @@
...
 
 ## Features
 
-Sui full nodes exist to:
+Sui Full nodes:
 
 * Track and verify the state of the blockchain, independently and locally.
 * Serve read requests from clients.
 * Conduct local app testing against verified data.
```
Comment (longbowlu) [inline doc/src/build/fullnode.md:16]: what is `Conduct local app testing against verified data.`?

File: doc/src/build/fullnode.md [hunk 5]
```diff
@@ -1,58 +1,47 @@
...
 ## State synchronization
 
-Today Sui full nodes sync with validators to be able to learn about newly
-committed transactions.
+Sui Full nodes sync with validators to receive newly committed transactions.
 
-The normal life of a transaction requires a few round trips to 2f+1 validators
-to be able to form a TxCert, at which point a transaction is guaranteed to be
-committed and executed.
+A transaction requires a few round trips to 2f+1 validators to be able to form a TxCert, at which point a transaction is guaranteed to be committed and executed.
```
Comment (longbowlu) [inline doc/src/build/fullnode.md:22]: better to use full name for `TxCert` ?

`at which point a transaction is guaranteed to be committed and executed.` is not 100% accurate as well. I think this finality topic perhaps deserves a separate page

File: doc/src/build/fullnode.md [hunk 6]
```diff
@@ -1,58 +1,47 @@
...
 
-Today, this synchronization process is performed by:
+This synchronization process is performed by:
 
 1. Following 2f+1 validators and listening for newly committed transactions.
 1. Requesting the transaction from one validator.
```
Comment (longbowlu) [inline doc/src/build/fullnode.md:27]:
```suggestion
1. Making sure the transaction is recognized by 2f+1 validators and hence reaches finality.
```

File: doc/src/build/fullnode.md [hunk 7]
```diff
@@ -1,58 +1,47 @@
...
 1. Locally executing the transaction and updating the local DB.
 
 This synchronization process is far from ideal as it requires listening
-to at a minimum 2f+1 validators to ensure that a full node has properly seen all
+to at a minimum 2f+1 validators to ensure that a Full node has properly seen all
 new transactions. Over time, we will improve this process (e.g. with the
-introduction of checkpoints, ability to synchronize with other full nodes,
-etc.) in order to have better guarantees around a full node’s ability to be
+introduction of checkpoints, ability to synchronize with other Full nodes,
```
Comment (longbowlu) [inline doc/src/build/fullnode.md:33]: checkpoints is already there. cc @lxfind / @mystenmark  to provide a paragraph here

File: doc/src/build/fullnode.md [hunk 8]
```diff
@@ -61,23 +50,23 @@ Conversely, a validator needs to store only the latest transactions on the
...
+We recommend the following minimum hardware requirements for running a Full node:
 
 * CPUs: 2
 * RAM: 8GB
 * Storage: 50GB
 
 ### Software requirements
 
-We recommend running Sui full nodes on Linux. The Sui team supports the Ubuntu and
+We recommend running Sui Full nodes on Linux. The Sui team supports the Ubuntu and
```
Comment (longbowlu) [inline doc/src/build/fullnode.md:65]: "The Sui team supports" sounds a bit weird

File: doc/src/build/json-rpc.md [hunk 9]
```diff
@@ -27,38 +27,28 @@ Use the following command to start an RPC server:
 ```shell
 $ rpc-server
```
Comment (longbowlu) [inline doc/src/build/json-rpc.md:28]: `rpc-server` is equivalent to Gateway. we should remove relevant sections in this page and link to `fullnode.md` for how to run a fullnode

File: doc/src/build/json-rpc.md [hunk 10]
```diff
@@ -27,38 +27,28 @@ Use the following command to start an RPC server:
...
-```
-
 > **Note:** For additional logs, set `RUST_LOG=debug` before invoking `rpc-server`.
 
-Export a local user variable to store the hardcoded hostname + port that the local RPC server starts with to be used when issuing the `curl` commands that follow.
+Export a local user variable to store the hardcoded hostname + port for the local RPC server. You can then use it in `curl` commands.
+
 ```shell
 export SUI_RPC_HOST=http://127.0.0.1:5001
 ```
```
Comment (longbowlu) [inline doc/src/build/json-rpc.md:36]: these need to be removed as well

File: doc/src/build/sui-local-network.md [hunk 11]
```diff
@@ -0,0 +1,58 @@
...
+
+To create the configuration files and objects for a local Sui network, run the `genesis` command. Genesis creates the network configuration files in the ~/.sui/sui_config folder. This includes a YAML file for fullnode, network, client, and each validator. It also creates a sui.keystore that stores client key pairs. 
+
+The network that genesis creates includes four validators and five user accounts that contain five coin objects each.
+
+   ```shell
+   $ sui genesis
+   ```
+
+The first time you use the client CLI, it creates a client.yaml file. If you use the default values, it connects to a Sui Devnet Full node. When you run genesis to create a local network, if the .sui/sui_config folder contains a client.yaml file, the genesis process warns you that the folder must be empty. You can use the `--force` argument to replace the configuration files, or use `--working-dir` to specify a different directory for the network configuration files.
```
Comment (longbowlu) [inline doc/src/build/sui-local-network.md:21]: "If you use the default values, it connects to a Sui Devnet Full node." 

this is true when you directly run `sui client` without first running `sui genesis`. If you run `sui genesis` first, `client.yaml` will be override by config that points to the local network

File: doc/src/build/sui-local-network.md [hunk 12]
```diff
@@ -0,0 +1,58 @@
...
+Use the following command to use a different directory to store the configuration files.
+```shell
+$ sui genesis --working-dir /workspace/config-files
+```
+
+The directory must already exist, and be empty, before you run the command.
+
+#### Embedded gateway
+
+You can use an embedded gateway with your local network. The gateway.yaml file contains information about the embedded gateway.
```
Comment (longbowlu) [inline doc/src/build/sui-local-network.md:37]: let's note here that embedded gateway will soon be going away too

File: doc/src/learn/how-sui-works.md [hunk 13]
```diff
@@ -2,13 +2,11 @@
...
 
 ## tl;dr
 
-The Sui blockchain operates at a speed and scale previously thought unimaginable. Sui assumes the typical blockchain transaction is a simple transfer and optimizes for that use. Sui does this by making each request idempotent, holding network connections open longer, and ensuring transactions complete immediately. Sui optimizes for single-writer objects, allowing a design that forgoes consensus for simple transactions.
+The Sui blockchain operates at a speed and scale previously thought unattainable. Sui assumes the typical blockchain transaction is a simple transfer and optimizes for that use. Sui does this by making each request idempotent, holding network connections open longer, and ensuring transactions complete immediately. Sui optimizes for single-writer objects, allowing a design that forgoes consensus for simple transactions.
```
Comment (longbowlu) [inline doc/src/learn/how-sui-works.md:9]: I think this paragraph is not true (any more): 
"Sui assumes the typical blockchain transaction is a simple transfer and optimizes for that use" 

We can probably say "The Sui blockchain operates at a speed and scale previously thought unattainable. Sui assumes most blockchain transactions touch non-overlapping states and thus can run in parallel with one another. Sui optimizes for single-writer objects, allowing a design that forgoes consensus for simple transactions."


> Sui does this by making each request idempotent, holding network connections open longer, and ensuring transactions complete immediately.

I don't know what this means lol

File: doc/src/learn/how-sui-works.md [hunk 14]
```diff
@@ -2,13 +2,11 @@
...
 
 Instead of the traditional blockchain’s fire-and-forget broadcast, Sui ensures a two-way handshake between the requestor and approving validators, with simple transactions having near instant finality. With this low latency, transactions can easily be incorporated into games and other settings that need completion in real time. Furthermore, Sui supports smart contracts written in Move, a language designed for blockchains with strong inherent security and a more understandable programming model.
```
Comment (longbowlu) [inline doc/src/learn/how-sui-works.md:11]: The way it works is not really "two-way handshake". I suggest we say 

"Instead of the traditional blockchain’s fire-and-forget broadcast, Sui's design enables a requestor, or a proxy to proactively talk to validators and bring a transaction to finality, with simple transactions having near instant finality.

File: doc/src/learn/sui-glossary.md [hunk 15]
```diff
@@ -6,7 +6,7 @@ Find terms used in Sui defined below. Where possible, we link to a canonical def
 
 ### Accumulator
```
Comment (longbowlu) [inline doc/src/learn/sui-glossary.md:7]: do we really use this term in sui docs?

File: doc/src/build/fullnode.md (hunk 4 above)
Comment (randall-Mysten) [reply doc/src/build/fullnode.md:16]: Unsure, this was existing content. Will remove

File: doc/src/build/json-rpc.md (hunk 9 above)
Comment (randall-Mysten) [reply doc/src/build/json-rpc.md]: updated

File: doc/src/build/json-rpc.md (hunk 10 above)
Comment (randall-Mysten) [reply doc/src/build/json-rpc.md]: removed

File: doc/src/build/sui-local-network.md (hunk 11 above)
Comment (randall-Mysten) [reply doc/src/build/sui-local-network.md]: added more info to clarify this detail

File: doc/src/build/sui-local-network.md (hunk 12 above)
Comment (randall-Mysten) [reply doc/src/build/sui-local-network.md]: added.

File: doc/src/learn/how-sui-works.md (hunk 13 above)
Comment (randall-Mysten) [reply doc/src/learn/how-sui-works.md]: nor I ;)
Updated the paragraph

File: doc/src/learn/how-sui-works.md (hunk 14 above)
Comment (randall-Mysten) [reply doc/src/learn/how-sui-works.md]: updated

File: doc/src/learn/sui-glossary.md (hunk 15 above)
Comment (randall-Mysten) [reply doc/src/learn/sui-glossary.md]: not anymore!

File: doc/src/build/json-rpc.md [hunk 16]
```diff
@@ -69,20 +40,30 @@ curl --location --request POST $SUI_RPC_HOST \
...
-A transaction data response will be returned from the gateway server.
+--data-raw '{
+  "jsonrpc": "2.0",
+  "id": 1,
+  "method": "sui_transferObject",
+  "params": [ 
+    {{tx_bytes}},
+    {{sig_scheme}},
+    {{signature}},
+    {{pub_key}},
```
[the remark this answers is not in the document]
Comment (randall-Mysten) [reply doc/src/build/json-rpc.md:62]: Updated, thank you

**Review by patrickkuo — APPROVED** [review]
LGTM! Thanks!

File: doc/src/build/comms.md [hunk 17]
```diff
@@ -1,16 +1,14 @@
 ---
-title: Connect and Communicate with the Sui Network
+title: Connect to the Sui Network
 ---
 
-Now that you have [installed Sui](install.md), [connected to Devnet](../build/devnet.md), and started to [create smart contracts in Move](move/index.md) and [program Sui objects](programming-with-objects/index.md), it's time to let your apps talk to Sui.
-
 The pages in this section provide various options for communicating with Sui:
 
 * Set up your [Sui RPC server and use the Sui JSON-RPC API](json-rpc.md) to interact with a local Sui network.
```
[the remark this answers is not in the document]
Comment (randall-Mysten) [reply doc/src/build/comms.md:7]: removed

File: doc/src/build/cli-client.md [hunk 18]
```diff
@@ -98,68 +89,43 @@ USAGE:
...
-But the actual address values will most likely differ
-in your case (as will other values, such as object IDs, in the latter
-parts of this tutorial). Consequently, **do not copy and paste
-the actual command from this tutorial as they are unlikely to work for
-you verbatim**. Each time you create a config for the client, addresses
-and object IDs will be assigned randomly. Consequently, you cannot rely
-on copy-pasting commands that include these values, as they will be different
-between different users/configs.
-
 ### Active address
```
Comment (Brunakalline) [inline doc/src/build/cli-client.md:98]: 0x0add8f2310517557684d138555694bbca57c899e

File: doc/src/build/cli-client.md [hunk 19]
```diff
@@ -98,68 +89,43 @@ USAGE:
...
 ```
 
-Which will reveal an address resembling:
+The response to the request resembles the following:
 
 ```shell
 0x562f07cf6369e8d22dbf226a5bfedc6300014837
 ```
 
-Changing the default address is as easy as calling the `switch` command:
```
Comment (Brunakalline) [inline doc/src/build/cli-client.md:323]: 0x0add8f2310517557684d138555694bbca57c899e

File: doc/src/build/cli-client.md [hunk 20]
```diff
@@ -2,21 +2,19 @@
 title: Use Sui CLI to Start and Manage Network
 ---
```
Comment (Brunakalline) [inline doc/src/build/cli-client.md:3]: $ which sui
